src/d3d9/d3d9_mem.cpp

```diff
@@ -0,0 +1,350 @@
+#include "d3d9_mem.h"
+#include "../util/util_string.h"
+#include "../util/util_math.h"
+#include "../util/log/log.h"
+#include "../util/util_likely.h"
+#include <utility>
+
+#ifdef D3D9_ALLOW_UNMAPPING
+#include <sysinfoapi.h>
+#endif
+
+namespace dxvk {
+
+#ifdef D3D9_ALLOW_UNMAPPING
+  D3D9MemoryAllocator::D3D9MemoryAllocator() {
+    SYSTEM_INFO sysInfo;
+    GetSystemInfo(&sysInfo);
+    m_allocationGranularity = sysInfo.dwAllocationGranularity;
+  }
+
+  D3D9Memory D3D9MemoryAllocator::Alloc(uint32_t Size) {
+    std::lock_guard<dxvk::mutex> lock(m_mutex);
+
+    uint32_t alignedSize = align(Size, 64);
+    for (auto& chunk : m_chunks) {
+      D3D9Memory memory = chunk->Alloc(alignedSize);
+      if (memory) {
+        m_usedMemory += memory.GetSize();
+        return memory;
+      }
+    }
+
+    alignedSize = align(Size, 4096);
```

Comment:
If this is 4096 to match the page size, can we use `dwPageSize` from the system info struct instead?

Reply:
This is a leftover from when I started this. I expected this API to require page aligned sizes but I dont think it has that requirement. So I'll just remove this line.